aea/decision_maker/base.py

```diff
@@ -345,24 +343,22 @@ def _is_acceptable_tx(self, crypto_object: Crypto, amount: float) -> bool:
         :param amount: the tx amount
         :return: whether the transaction is acceptable or not
         """
-        balance = cast(float, crypto_object.token_balance)
-        affordable = amount <= balance
+        balance = self.ledger_apis.token_balance(crypto_object.identifier, cast(str, crypto_object.address))
+        affordable = amount <= cast(float, balance)
         # TODO check against preferences and other constraints
         return affordable
 
-    def _settle_tx(self, crypto_object: Crypto, counterparty_pbk: str, amount: float, tx_fee: float) -> str:
+    def _settle_tx(self, crypto_object: Crypto, counterparty_address: str, amount: float, tx_fee: float) -> Optional[str]:
         """
         Settle the tx.
 
         :param crypto_object: the crypto object
-        :param counterparty_pbk: the counterparty pbk
+        :param counterparty_address: the counterparty address
         :param amount: the tx amount
         :param tx_fee: the tx fee
         :return: the transaction digest
         """
-        counterparty_address = crypto_object.generate_counterparty_address(counterparty_pbk)
-        crypto_object.transfer(counterparty_address, amount, tx_fee)
-        tx_digest = 'something'  # TODO: get tx_digest from the ledger api
+        tx_digest = self.ledger_apis.transfer(crypto_object.identifier, crypto_object.entity, counterparty_address, amount, tx_fee)
```

Comment:
minor comment: the argument `crypto_object.entity` gives me:

`Expected type: Entity, got '() -> Any'`.

So I suggest two changes to `aea.crypto.base.Crypto.entity`:
- make it a property
- the return value of the property is `fetchai.ledger.crypto.Entity`, or an `Union[Entity]` such that in the future we can extend it with other kinds of entities.